components/footer/Footer.tsx

```diff
@@ -1,173 +1,284 @@
+import ColorClasses from "$store/components/footer/ColorClasses.tsx";
+import Divider from "$store/components/footer/Divider.tsx";
+import BackToTop from "$store/components/footer/BackToTop.tsx";
+import type { Image as LiveImage } from "deco-sites/std/components/types.ts";
 
-export type IconItem = { icon: AvailableIcons };
-export type StringItem = {
+export type Item = {
   label: string;
   href: string;
 };
 
-export type Item = StringItem | IconItem;
-
 export type Section = {
   label: string;
-  children: Item[];
+  items: Item[];
 };
 
-const isIcon = (item: Item): item is IconItem =>
-  // deno-lint-ignore no-explicit-any
-  typeof (item as any)?.icon === "string";
+export interface SocialItem {
+  label:
+    | "Discord"
+    | "Facebook"
+    | "Instagram"
+    | "Linkedin"
+    | "Tiktok"
+    | "Twitter";
+  link: string;
+}
 
-function SectionItem({ item }: { item: Item }) {
-  return (
-    <span class="text-primary-content">
-      {isIcon(item)
-        ? (
-          <div class="border-base-100 border border-solid py-1.5 px-2.5">
-            <Icon
-              id={item.icon}
-              width={25}
-              height={20}
-              strokeWidth={0.01}
-            />
-          </div>
-        )
-        : (
-          <a href={item.href}>
-            {item.label}
-          </a>
-        )}
-    </span>
-  );
+export interface PaymentItem {
+  label: "Diners" | "Elo" | "Mastercard" | "Pix" | "Visa";
+}
+
+export interface MobileApps {
+  /** @description Link to the app */
+  apple?: string;
+  /** @description Link to the app */
+  android?: string;
 }
 
-function FooterContainer(
-  { children, class: _class = "" }: {
-    class?: string;
-    children: ComponentChildren;
-  },
-) {
-  return <div class={`py-6 px-4 sm:py-12 sm:px-0 ${_class}`}>{children}</div>;
+export interface RegionOptions {
+  currency?: Item[];
+  language?: Item[];
+}
+
+export interface NewsletterForm {
+  placeholder?: string;
+  buttonText?: string;
+  /** @format html */
+  helpText?: string;
 }
 
 export interface Props {
+  logo?: {
+    image: LiveImage;
+    description?: string;
+  };
+  newsletter?: {
+    title?: string;
+    /** @format textarea */
+    description?: string;
+    form?: NewsletterForm;
+  };
   sections?: Section[];
+  social?: {
+    title?: string;
+    items: SocialItem[];
+  };
+  payments?: {
+    title?: string;
+    items: PaymentItem[];
+  };
+  mobileApps?: MobileApps;
+  regionOptions?: RegionOptions;
+  extraLinks?: Item[];
+  backToTheTop?: {
+    text?: string;
+  };
+  layout?: {
+    backgroundColor?:
+      | "Primary"
+      | "Secondary"
+      | "Accent"
+      | "Base 100"
+      | "Base 100 inverted";
+    variation?:
+      | "Variation 1"
+      | "Variation 2"
+      | "Variation 3"
+      | "Variation 4"
+      | "Variation 5";
+    hide?: {
+      logo?: boolean;
+      newsletter?: boolean;
+      sectionLinks?: boolean;
+      socialLinks?: boolean;
+      playmentMethods?: boolean;
```

Comment:
typo, pay attention that you must change in all occurences
```suggestion
      paymentMethods?: boolean;
```